tasks/post_to_catalog/post_to_catalog.py

```diff
@@ -0,0 +1,241 @@
+def send_record_to_database(record: Dict[str, Any], engine: Engine) -> None:
+    """
+    Deconstructs a record to its components and calls create_catalog_records with the result.
+
+    Args:
+        record: Contains the following keys:
+            'body' (str): A json string representing a dict.
+                Contains key/value pairs of column names and values for those columns.
+                Must match catalog_record_input.json.
+        engine: The sqlalchemy engine to use for contacting the database.
+    """
+    values = json.loads(record["body"])
+    _CATALOG_RECORD_VALIDATE(values)
+    create_catalog_records(
+        values["provider"],
+        values["collection"],
+        values["granule"],
+        engine,
+    )
+
+
+def create_catalog_records(
+    provider: Dict[str, str],
+    collection: Dict[str, str],
+    granule: Dict[str, Union[str, List[Dict[str, Union[str, int]]]]],
+    engine: Engine,
+) -> None:
+    """
+    Posts the information to the catalog database.
+
+    Args:
+        provider: See schemas/catalog_record_input.json.
+        collection: See schemas/catalog_record_input.json.
+        granule: See schemas/catalog_record_input.json.
+        engine: The sqlalchemy engine to use for contacting the database.
+    """
+    try:
+        LOGGER.debug("Creating catalog records.")
+        with engine.begin() as connection:
+            LOGGER.debug("Creating provider record '{providerId}'", providerId=provider["providerId"])
+            connection.execute(
+                create_provider_sql(),
+                [
+                    {
+                        "provider_id": provider["providerId"],
+                        "name": provider["name"],
+                    }
+                ],
+            )
+            LOGGER.debug("Creating collection record '{collectionId}'", collectionId=collection["collectionId"])
+            connection.execute(
+                create_collection_sql(),
+                [
+                    {
+                        "collection_id": collection["collectionId"],
+                        "shortname": collection["shortname"],
+                        "version": collection["version"],
+                    }
+                ],
+            )
+            LOGGER.debug("Creating granule record '{cumulusGranuleId}'", cumulusGranuleId=granule["cumulusGranuleId"])
+            results = connection.execute(
+                create_granule_sql(),
+                [
+                    {
+                        "collection_id": collection["collectionId"],
+                        "cumulus_granule_id": granule["cumulusGranuleId"],
+                        "execution_id": granule["executionId"],
+                        "ingest_time": granule["ingestTime"],
+                        "cumulus_create_time": granule["cumulusCreateTime"],
+                        "last_update": granule["lastUpdate"],
+                    }
+                ],
+            )
+
+            for (
```

Comment:
Not sure about this logic here. We may be able to simplify this to a dictionary and return as a 1 liner since this is a ResultProxy if I recall. Maybe something like this?
```
results_dict = [{column: value for column, value in row.items()} for row in results]
granule_id = results_dict[0].id
```